xmake/modules/private/utils/batchcmds.lua

```diff
@@ -231,8 +231,8 @@ function batchcmds:compile(sourcefiles, objectfile, opt)
     local program, argv = compiler_inst:compargv(sourcefiles, objectfile, opt)
 
     -- add compilation command and bind run environments of compiler
-    self:mkdir(path.directory(objectfile))
-    self:vrunv(program, argv, {envs = table.join(compiler_inst:runenvs(), opt.envs)})
+    self:mkdir(path(path.directory(objectfile)))
```

Comment:
we need not add `path()` for mkdir, because it is always path. we can translate it's path in vs201x directly.

so we need only wrap `path` in execv